### PR TITLE
Output a warning on mulitple paths being tested in the same file.

### PR DESCRIPTION
**Pull request by dblock**
### Description

Adds support for story warnings, outputs a warning when multiple paths are being tested.

```
$ npm run test:spec--insecure -- --tests tests/indices/

> opensearch_api_tools@1.0.0 test:spec--insecure
> ts-node tools/src/tester/test.ts --opensearch-insecure --tests tests/indices/

OpenSearch 2.15.0

PASSED  cache.yaml (/Users/dblock/source/opensearch-project/opensearch-api-specification/dblock-opensearch-api-specification/tests/indices/cache.yaml)
PASSED  clone.yaml (/Users/dblock/source/opensearch-project/opensearch-api-specification/dblock-opensearch-api-specification/tests/indices/clone.yaml)
WARNING Multiple paths detected, please group similar tests together and move paths not being tested to the prologue.
  /{index}/_settings
  /{index}/_clone/{target}
```

This lets us catch the instances where we maybe be testing multiple/unrelated paths or where some should be in prologue. It helps make tests more contained, and warnings can be suppressed when absolutely necessary.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: json_schemas/test_story.schema.yaml [hunk 1]
```diff
@@ -163,3 +165,11 @@ definitions:
       - type: string
       - type: number
       - type: boolean
+
+  Warnings:
+    type: object
+    properties:
+      multiple-paths-detected:
+        type: boolean
+        default: true
```
Comment (nhtruong) [inline json_schemas/test_story.schema.yaml:174]: It's unclear what turning this off and on means. Please add a description for this property.

File: tools/src/tester/StoryEvaluator.ts [hunk 2]
```diff
@@ -51,7 +52,27 @@ export default class StoryEvaluator {
...
+
+  #warning_if_mismatched_chapter_paths(chapters: Chapter[]): string | undefined {
+    const paths = _.compact(_.map(chapters, (chapter) => {
+      const multiple_paths_detected = chapter.warnings?.['multiple-paths-detected'] ?? true
+      if (multiple_paths_detected) return chapter.path
+    }))
+    const normalized_paths = _.map(paths, (path) => path.replaceAll(/\/\{[^}]+}/g, '').replaceAll('//', '/'))
+    const paths_counts: Record<string, number> = Object.assign((_.values(_.groupBy(normalized_paths)).map(p => { return { [p[0]] : p.length } })))
+    if (paths_counts.length > 1) {
+      return `Multiple paths detected, please group similar tests together and move paths not being tested to the prologue.\n  ${_.join(_.uniq(paths), "\n  ")}\n`
```
Comment (nhtruong) [inline tools/src/tester/StoryEvaluator.ts:75]: nit: prologue --> prologues
It's possible that the untested paths cold be  moved to the epilogues too right?

Comment (dblock) [reply tools/src/tester/StoryEvaluator.ts:75]: Yes, updated.

**Review by nhtruong — DISMISSED** [review]

**Comment by dblock** [comment]
@nhtruong rebased

**Pull request merged**